sdk/storage/Azure.Storage.Blobs/src/PartitionedDownloader.cs

```diff
@@ -0,0 +1,203 @@
+                ContentLanguage = new string[] { response.Value.Details.ContentLanguage },
+
+                // TODO: These do not exist on BlobDownloadInfo
+                //AccessTier = response.Value.Details.AccessTier,
+                //AccessTierInferred = response.Value.Details.AccessTierInferred,
+                //ArchiveStatus = response.Value.Details.ArchiveStatus,
+                //AccessTierChangedOn = response.Value.Details.AccessTierChangedOn,
+                //IsIncrementalCopy = response.Value.Details.IsIncrementalCopy,
+                //DestinationSnapshot = response.Value.Details.DestinationSnapshot,
+                //CreatedOn = response.Value.Details.CreatedOn,
+            }, response.GetRawResponse());
+        }
+
+        public Response<BlobProperties> DownloadTo(Stream stream, BlobRequestConditions conditions, CancellationToken cancellationToken)
+        {
+            var initialRange = new HttpRange(0, _singleBlockThreshold);
+            Response<BlobDownloadInfo> initialRequest = _client.Download(initialRange, conditions, rangeGetContentHash: false, cancellationToken);
+
+            long initialLength = initialRequest.Value.ContentLength;
+            long totalLength = ParseLength(initialRequest.Value.Details.ContentRange);
+            ETag etag = initialRequest.Value.Details.ETag;
+
+            Response<BlobProperties> properties = CreateProperties(initialRequest, totalLength);
+
+            CopyTo(initialRequest, stream, cancellationToken);
+
+            if (initialLength == totalLength)
+            {
+                return properties;
+            }
+
+            BlobRequestConditions conditionsWithEtag = CreateConditionsWithEtag(conditions, etag);
+
+            foreach (HttpRange httpRange in GetRanges(initialLength, totalLength))
+            {
+                BlobDownloadInfo result = _client.Download(httpRange, conditionsWithEtag, rangeGetContentHash: false, cancellationToken);
+
+                CopyTo(result, stream, cancellationToken);
+            }
+
+            return properties;
+        }
+
+        private static BlobRequestConditions CreateConditionsWithEtag(BlobRequestConditions conditions, ETag etag)
+        {
+            BlobRequestConditions conditionsWithEtag = new BlobRequestConditions()
+            {
+                LeaseId = conditions?.LeaseId,
+                IfMatch = conditions?.IfMatch ?? etag,
+                IfNoneMatch = conditions?.IfNoneMatch,
+                IfModifiedSince = conditions?.IfModifiedSince,
+                IfUnmodifiedSince = conditions?.IfUnmodifiedSince
+            };
+            return conditionsWithEtag;
+        }
+
+        private static async Task CopyToAsync(BlobDownloadInfo result, Stream stream, CancellationToken cancellationToken)
+        {
+            await result.Content.CopyToAsync(stream, 8096, cancellationToken).ConfigureAwait(false);
```

Comment:
Nit - add a value for `8096` in `Constants` (unless you want to use the existing `Constants.DefaultStreamCopyBufferSize == 81920`).